### PR TITLE
refactor[Wave]: CC => FC

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

Use more react way which support multiple wave motion:

Kapture 2022-12-28 at 16 48 51

### 📝 Changelog

- 将 Wave 重构为 Function Component

- 删除了通过 forwardRef 转发 ref，因为这个只在 test case 中用到了，在实际代码中是无用的

- 删除了test case 中关于 class 静态属性的测试用例，在 FC 中没办法获取到组件内部的属性

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Refactor wave effect that can now trigger multiple wave in one time. |
| 🇨🇳 Chinese | 重构水波纹视效，现在可以多个水波纹同时触发了。 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/_util/wave/index.tsx [hunk 1]
```diff
@@ -187,75 +154,91 @@ export class InternalWave extends React.Component<WaveProps> {
...
+          ? composeRef((children as any).ref, containerRef)
+          : containerRef;
+        return cloneElement(children, { ref });
+      }}
+    </ConfigConsumer>
+  );
+};
 
-const Wave = forwardRef<InternalWave, WaveProps>((props, ref) => {
+const Wave: React.FC<WaveProps> = (props) => {
```
Comment (MadCcc) [inline components/_util/wave/index.tsx:239]: 这个 wrapper 在 v5 特地拆出来的，可以不要了，合进去

Comment (li-jia-nan) [reply components/_util/wave/index.tsx]: done

File: components/_util/wave/index.tsx [hunk 2]
```diff
@@ -187,75 +154,86 @@ export class InternalWave extends React.Component<WaveProps> {
...
+      if (clickWaveTimeoutId.current) {
+        clearTimeout(clickWaveTimeoutId.current);
+      }
+    };
+  }, []);
   useStyle();
-  return <InternalWave ref={ref} {...props} />;
-});
+  return (
+    <ConfigConsumer>
```
Comment (MadCcc) [inline components/_util/wave/index.tsx:224]: 用 useContext 吧

Comment (li-jia-nan) [reply components/_util/wave/index.tsx]: done

**Comment by li-jia-nan** [comment]
删了几个 case，cov 果然掉了……

**Comment by MadCcc** [comment]
原则上不删已有的 case，特别是重构

**Comment by li-jia-nan** [comment]
唉？merge 了一下 master 分支，modal 组件的 case 怎么也挂了……

**Comment by zombieJ** [comment]
`components/_util/raf.ts` 这个看起来和 `rc-util` 的 `raf` 是等价的，我删了试试。

**Review by zombieJ — APPROVED** [review]

**Pull request merged**